dataland-community-manager/src/test/kotlin/org/dataland/datalandcommunitymanager/services/CommunityManagerListenerUnitTest.kt

```diff
@@ -0,0 +1,220 @@
+package org.dataland.datalandcommunitymanager.services
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import org.dataland.datalandbackend.openApiClient.model.DataTypeEnum
+import org.dataland.datalandbackend.openApiClient.model.SourceabilityInfo
+import org.dataland.datalandbackendutils.model.QaStatus
+import org.dataland.datalandmessagequeueutils.constants.ActionType
+import org.dataland.datalandmessagequeueutils.constants.MessageType
+import org.dataland.datalandmessagequeueutils.exceptions.MessageQueueRejectException
+import org.dataland.datalandmessagequeueutils.messages.PrivateDataUploadMessage
+import org.dataland.datalandmessagequeueutils.messages.QaStatusChangeMessage
+import org.dataland.datalandmessagequeueutils.messages.SourceabilityMessage
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.CsvSource
+import org.junit.jupiter.params.provider.EnumSource
+import org.mockito.kotlin.any
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.reset
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+
+/**
+ * Tests if the listener processes the incoming non-sourceable data information correctly.
+ */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class CommunityManagerListenerUnitTest {
+    private lateinit var communityManagerListener: CommunityManagerListener
+    private val jacksonObjectMapper = jacksonObjectMapper().findAndRegisterModules()
+    private val mockDataRequestUpdateManager = mock<DataRequestUpdateManager>()
+    private val mockInvestorRelationsManager = mock<InvestorRelationsManager>()
+    private val validDataId = "valid-data-id"
+    private val invalidDataId = ""
+    private val correlationId = "test correlation id"
+
+    private val typeQAStatusChange = MessageType.QA_STATUS_UPDATED
+    private val typePrivateUpload = MessageType.PRIVATE_DATA_RECEIVED
+    private val typeNonSourceable = MessageType.DATA_NONSOURCEABLE
+
+    @BeforeEach
+    fun setUp() {
+        reset(
+            mockDataRequestUpdateManager,
+            mockInvestorRelationsManager,
+        )
+        communityManagerListener =
+            CommunityManagerListener(
+                jacksonObjectMapper,
+                mockDataRequestUpdateManager,
+                mockInvestorRelationsManager,
+            )
+    }
+
+    @ParameterizedTest
+    @EnumSource(QaStatus::class)
+    fun `valid QA status change message should be dealt with appropriately`(qaStatus: QaStatus) {
+        val qaStatusChangeMessage: QaStatusChangeMessage
+        when (qaStatus) {
```

Comment:
Der ganze Block kann weg:

```suggestion
        if (qaStatus == QaStatus.Pending) return
        val qaStatusChangeMessage = QaStatusChangeMessage(dataId = validDataId, updatedQaStatus = qaStatus, currentlyActiveDataId = validDataId)
```

Reply:
Done.